test/interpreter/function-now.spec.ts

```diff
@@ -9,17 +9,62 @@ function sleepFor( sleepDuration: number ){
 }
 
 describe('Interpreter - function NOW', () => {
+  let originalNow: () => number
+
+  beforeEach(() => {
+    originalNow = Date.now
+    let cnt = 20
+    Date.now = () => {
+      cnt += 1
+      return Date.parse(`1985-08-16T03:45:${cnt}`)
+    }
+  })
+
   it('works',  () => {
     const engine =  HyperFormula.buildFromArray([
       ['=NOW()'],
     ])
     const t1 = engine.getCellValue(adr('A1')) as number
+    expect(t1).toBeCloseTo(31275.1565856481)
     expect(engine.getCellValueDetailedType(adr('A1'))).toBe(CellValueDetailedType.NUMBER_DATETIME)
-    sleepFor(1000)
     engine.setCellContents(adr('A2'), null)
-    const t2 = engine.getCellValue(adr('A1')) as number
-    const delta = (t2-t1)*(24*60*60)
-    expect(delta).toBeCloseTo(1)
+    const delta = engine.getCellValue(adr('A1')) as number - t1
+    expect(delta*24*60*60).toBeGreaterThanOrEqual(1) //internals of the engine are also using Date.now(), so the value should be actually 4 or even more
```

Comment:
Wouldn't it be better to have this test predictable?

```ts
dateMock.mockReturnValueOnce(new Date('1985-08-16T03:45:30Z').getTime())

const t1 = engine.getCellValue(adr('A1'))

dateMock.mockReturnValueOnce(new Date('1985-08-16T03:45:30Z').getTime())

engine.setCellContents(adr('A2'), null)
const t2 = engine.getCellValue(adr('A1'))
```

we will know exactly what time each `NOW()` returns and can make precise `expect`ation